### PR TITLE
optionally disable header heuristics

### PR DESCRIPTION
**Pull request by de-code**
see discussion in #476

/cc @lfoppiano @kermitt2

### CODE REVIEW

File: grobid-core/src/main/java/org/grobid/core/engines/Engine.java [hunk 1]
```diff
@@ -372,8 +372,12 @@ public String processHeader(String inputFile, GrobidAnalysisConfig config, Bibli
...
 
-        Pair<String, Document> resultTEI = parsers.getHeaderParser().processing2(inputFile, result, config);
-		//Pair<String, Document> resultTEI = parsers.getHeaderParser().processing(inputFile, result, config);
+        Pair<String, Document> resultTEI;
+        if (GrobidProperties.isHeaderUseHeuristics()) {
+            resultTEI = parsers.getHeaderParser().processing2(inputFile, result, config);
+        } else {
+            resultTEI = parsers.getHeaderParser().processing(new File(inputFile), result, config);
+        }
         Document doc = resultTEI.getRight();
```
Comment (de-code) [inline grobid-core/src/main/java/org/grobid/core/engines/Engine.java]: what is the purpose of calling `getRight`? We don't seem to use `doc`.

File: grobid-core/src/main/java/org/grobid/core/engines/Engine.java [hunk 2]
```diff
@@ -372,8 +372,12 @@ public String processHeader(String inputFile, GrobidAnalysisConfig config, Bibli
             result = new BiblioItem();
         }
 
-        Pair<String, Document> resultTEI = parsers.getHeaderParser().processing2(inputFile, result, config);
-		//Pair<String, Document> resultTEI = parsers.getHeaderParser().processing(inputFile, result, config);
+        Pair<String, Document> resultTEI;
+        if (GrobidProperties.isHeaderUseHeuristics()) {
+            resultTEI = parsers.getHeaderParser().processing2(inputFile, result, config);
```
Comment (de-code) [inline grobid-core/src/main/java/org/grobid/core/engines/Engine.java]: I am scared to rename `processing2` and `processing`..

File: grobid-core/src/main/java/org/grobid/core/engines/Engine.java (hunk 1 above)
Comment (kermitt2) [reply grobid-core/src/main/java/org/grobid/core/engines/Engine.java]: very likely some old stuff not used any more!

**Comment by kermitt2** [comment]
We want to allow by default the heuristics, no? It performs better in average and most of the users will still benefit from it until some efforts are made on the segmentation model.  
Not using the heuristics would be more at this stage for the power users without fear !

**Comment by de-code** [comment]
> We want to allow by default the heuristics, no? It performs better in average and most of the users will still benefit from it until some efforts are made on the segmentation model.
> Not using the heuristics would be more at this stage for the power users without fear !

Okay, I think I misunderstood you. You seemed keen to disable it all together. Leaving it as it is now is the safer option. I will change the default.

**Comment by de-code** [comment]
I was going to add `Engine` and `FullTextParser` tests but I think it would be easier if they were broken up more. Probably not worth it.

**Comment by kermitt2** [comment]
May I merge now ?

**Comment by de-code** [comment]
> May I merge now ?

Yes, please. That would be a new record :D

**Pull request merged**